includes/embeds/class-amp-base-embed-handler.php

```diff
@@ -107,4 +163,90 @@ function ( $attr_name ) {
 	}
+
+	/**
+	 * Get all child elements of the specified element.
+	 *
+	 * @param DOMElement $node Element.
+	 * @return DOMElement[] Array of child elements for specified element.
+	 */
+	protected function get_child_elements( DOMElement $node ) {
+		return array_filter(
+			iterator_to_array( $node->childNodes ),
+			static function ( DOMNode $child ) {
+				return $child instanceof DOMElement;
+			}
+		);
+	}
+
+	/**
+	 * Replace the node's parent with itself if the parent is a <p> tag, has no attributes and has no other children.
+	 * This usually happens while `wpautop()` processes the element.
+	 *
+	 * @since 1.6
+	 *
+	 * @param DOMElement $node Node.
+	 */
+	protected function unwrap_p_element( DOMElement $node ) {
+		$parent_node = $node->parentNode;
+		while ( $parent_node && ! ( $parent_node instanceof DOMElement ) ) {
+			$parent_node = $parent_node->parentNode;
+		}
+
+		if ( $parent_node instanceof DOMElement && 'p' === $parent_node->nodeName && false === $parent_node->hasAttributes() ) {
+			$child_element_count = count( $this->get_child_elements( $parent_node ) );
+			if ( 1 === $child_element_count ) {
+				$parent_node->parentNode->replaceChild( $node, $parent_node );
+			}
+		}
+	}
+
+	/**
+	 * Removes the node's nearest <script> sibling with a `src` attribute containing the base `src` URL provided.
+	 *
+	 * @since 1.6
+	 *
+	 * @param DOMElement $node         The DOMNode to whose sibling is the script to be removed.
+	 * @param string     $base_src_url Script URL to match against.
+	 * @param string     $content      Text content of node to match against.
+	 * @param bool       $is_next_to   Whether the script sibling is next or preceding the specified element.
+	 */
+	protected function remove_script_sibling( DOMElement $node, $base_src_url, $content = '', $is_next_to = true ) {
+		$sibling_location = $is_next_to ? 'nextSibling' : 'previousSibling';
+		$element_sibling  = $node->{$sibling_location};
+
+		while ( $element_sibling && ! ( $element_sibling instanceof DOMElement ) ) {
+			$element_sibling = $element_sibling->{$sibling_location};
+		}
+
+		// Handle case where script is wrapped in paragraph by wpautop.
+		if ( $element_sibling instanceof DOMElement && 'p' === $element_sibling->nodeName ) {
+			$children_elements = array_values( $this->get_child_elements( $element_sibling ) );
+
+			if (
+				1 === count( $children_elements ) &&
+				'script' === $children_elements[0]->nodeName &&
+				(
+					( $base_src_url && false !== strpos( $children_elements[0]->getAttribute( 'src' ), $base_src_url ) ) ||
+					( $content && false !== strpos( $children_elements[0]->textContent, $content ) )
+				)
+			) {
+				$element_sibling->parentNode->removeChild( $element_sibling );
+				return;
+			}
+		}
+
+		// Handle case where script is immediately following.
+		$is_embed_script = (
+			$element_sibling instanceof DOMElement &&
+			'script' === strtolower( $element_sibling->nodeName ) &&
```

Comment:
Is `strtolower()` needed here? It's not used above:

```suggestion
			'script' === $element_sibling->nodeName &&
```

I found that in all PHP versions the value is normalized to lower case already: https://3v4l.org/oDCO6